pyccl/base/schema.py

```diff
@@ -136,6 +136,27 @@ def Funlock(cls, cl, name, mutate: bool):
             newfunc = cls.unlock_instance(mutate=mutate)(func)
             setattr(cl, name, newfunc)
 
+    # TODO: Remove for CCLv3 & replace with Funlock.
+    @classmethod
+    def Funlock2(cls, cl, name):
+        """Replacement for Funlock incompatibility with signature.bind."""
+
+        def lock_on_exit(func):
+            @functools.wraps(func)
+            def wrapper(self, *args, **kwargs):
+                out = func(self, *args, **kwargs)
+                init_name = self.__class__.__init__.__qualname__
+                this_name = func.__qualname__
+                if this_name is init_name:  # skip this if called via `super()`
+                    self._object_lock.lock()
+                return out
+            return wrapper
+
+        func = vars(cl).get(name)
+        if func is not None:
+            newfunc = lock_on_exit(func)
+            setattr(cl, name, newfunc)
+
```

Comment:
At this point I've lost track of how many times we've lost and brought back these functions. Please explain why we need to bring this back (or in)

Reply:
bad merge, fixed, this shouldn't be there